# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/faith-amanze/content-refresh-prioritization/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing the Content Refresh Prioritization lane. In notebook 01, a hand-written rule scored about 0.68 precision at identifying pages worth reviewing, while a simple decision tree matched or beat it (0.60-1.00 depending on depth). That gap between a simple rule and a learned model suggests there's real signal in this data worth exploring further, specifically around what makes a page "stale" and worth refreshing, and whether a model can flag the right pages more reliably than a hand rule.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

This work supports a content team's decision about which pages to prioritize for review and refresh. The unit of analysis is a single page.

Decision: which pages should go into this week's refresh queue, out of thousands of candidates.

Who acts on it: a content strategist or SEO analyst, who pulls the top-ranked pages from the model's output and manually reviews or updates them.

Action: the team spends limited review hours on the pages the model flags as highest priority, rather than reviewing pages at random or by gut feel.

Cost of a wrong call: this cuts both ways. A false negative (a genuinely stale, underperforming page the model ranks low) means it gets skipped, and the business keeps losing traffic or clicks it could have recovered. A false positive (a fine page flagged as high priority) wastes scarce content-team hours on a page that didn't need attention. Since review time is the limited resource, false positives waste effort directly, while false negatives cost a harder-to-see opportunity, so both matter but in different ways.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/faith-amanze/content-refresh-prioritization/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

# Number 1: how many pages are in the dataset
print(f"Total pages in dataset: {len(df)}")

# Number 2: hand-rule baseline vs learned model precision
# (reference values carried over from notebooks/01_first_look_and_discovery.ipynb,
# not recomputed here -- that notebook is the source of truth for these two numbers)
print(f"Hand-rule baseline Precision@50 (from notebook 01): 0.68")
print(f"Decision tree Precision@50, depth=4, in-sample (from notebook 01): 1.00")

# Number 3: relationship between position and CTR
correlation = df['avg_position'].corr(df['ctr'])
print(f"Correlation between avg_position and ctr: {correlation:.3f}")

Total pages in dataset: 30000
Hand-rule baseline Precision@50 (from notebook 01): 0.68
Decision tree Precision@50, depth=4, in-sample (from notebook 01): 1.00
Correlation between avg_position and ctr: -0.073


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

Across the full dataset of 30,000 pages, a simple hand-written rule already achieves 0.68 precision at identifying pages worth reviewing, and a decision tree pushes that as high as 1.00 in-sample. That gap, even if partly inflated by overfitting on in-sample data, points to real learnable structure in what separates a page worth refreshing from one that isn't.

Interestingly, avg_position and ctr show only a very weak correlation (-0.073) across the whole dataset, which suggests position alone doesn't explain much of the variation in click-through rate here. That weak single-variable relationship is itself a reason a model-based approach may add more value than a simple rule: several weak signals combined might predict "worth refreshing" better than any one signal alone, which is exactly what this lane and the next 7 weeks would test.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.